src/main/java/seedu/address/model/person/Person.java

```diff
@@ -100,13 +100,23 @@ public boolean isSamePerson(Person otherPerson) {
         }
 
         boolean isSameName = otherPerson != null
-                && otherPerson.getName().getLowerCaseName().equals(getName().getLowerCaseName());
+                && otherPerson.getName().equals(getName());
         boolean isSameAddress = otherPerson != null
                 && otherPerson.getAddress().equals(getAddress());
 
         return isSameName && isSameAddress;
     }
 
+    /**
+     * Returns true if both persons have the same name in lower case.
+     * This defines a weaker notion of equality between two persons.
+     */
+    public boolean hasSimilarName(Person otherPerson) {
+        return otherPerson != null
+                && otherPerson.getName().getLowerCaseName()
+                .equals(getName().getLowerCaseName());
```

Comment:
Like the previous commit, should this be using `Name#equals` method instead?

Reply:
No, `Name#equals` is use for strict equality (i.e. same name (case-insensitive) and address) but this method is just for same name (case-insensitive).